code/modules/mob/living/simple_animal/slaughter/slaughter.dm

```diff
@@ -172,4 +52,10 @@ obj/effect/dummy/slaughter/relaymove(mob/user, direction)
 	desc = "It's still faintly beating with rage"
 	icon = 'icons/obj/surgery.dmi'
 	icon_state = "heart-on"
-	origin_tech = "combat=5;biotech=8"
\ No newline at end of file
+	origin_tech = "combat=5;biotech=8"
+
+/obj/item/weapon/demonheart/attack_self(mob/living/user)
+	visible_message("[user] feasts upon the [src].")
+	user << "You absorb some of the demon's power!"
+	user.bloodcrawl = 1
```

Comment:
If you've already eaten a heart, you should get bloodcrawl = 2, and if you're at bloodcrawl =2 already you shouldn't be able to eat it.